DataFormats/Portable/interface/PortableHostCollection.h

```diff
@@ -83,4 +84,232 @@ class PortableHostCollection {
+    return get<Idx>().view_;
+  }
+
+  template <std::size_t Idx = 0, typename = std::enable_if_t<(members_ > Idx)>>
+  ConstView<Idx> const& const_view() const {
+    return get<Idx>().view_;
+  }
+
+  template <std::size_t Idx = 0, typename = std::enable_if_t<(members_ > Idx)>>
+  View<Idx>& operator*() {
+    return get<Idx>().view_;
+  }
+
+  template <std::size_t Idx = 0, typename = std::enable_if_t<(members_ > Idx)>>
+  ConstView<Idx> const& operator*() const {
+    return get<Idx>().view_;
+  }
+
+  template <std::size_t Idx = 0, typename = std::enable_if_t<(members_ > Idx)>>
+  View<Idx>* operator->() {
+    return &get<Idx>().view_;
+  }
+
+  template <std::size_t Idx = 0, typename = std::enable_if_t<(members_ > Idx)>>
+  ConstView<Idx> const* operator->() const {
+    return &get<Idx>().view_;
+  }
+
+  // access the View by type
+  template <typename T>
+  typename T::View& view() {
+    return get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::ConstView const& view() const {
+    return get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::ConstView const& const_view() const {
+    return get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::View& operator*() {
+    return get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::ConstView const& operator*() const {
+    return get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::View* operator->() {
+    return &get<T>().view_;
+  }
+
+  template <typename T>
+  typename T::ConstView const* operator->() const {
+    return &get<T>().view_;
+  }
+
+  // access the Buffer
+  Buffer buffer() { return *buffer_; }
+  ConstBuffer buffer() const { return *buffer_; }
+  ConstBuffer const_buffer() const { return *buffer_; }
+
+  // Extract the sizes array
+  SizesArray sizes() const {
+    SizesArray ret;
+    constexpr_for<0, members_>([&](auto i) { ret[i] = get<i>().layout_.metadata().size(); });
+    return ret;
+  }
+  // part of the ROOT read streamer
+  static void ROOTReadStreamer(PortableHostMultiCollection* newObj, Implementation& onfileImpl) {
+    newObj->~PortableHostMultiCollection();
+    // use the global "host" object returned by cms::alpakatools::host()
+    std::array<int32_t, members_> sizes;
+    constexpr_for<0, members_>([&sizes, &onfileImpl](auto i) {
+      sizes[i] = static_cast<Leaf<i> const&>(onfileImpl).layout_.metadata().size();
+    });
+    new (newObj) PortableHostMultiCollection(sizes, cms::alpakatools::host());
+    constexpr_for<0, members_>([&newObj, &onfileImpl](auto i) {
+      static_cast<Leaf<i>&>(newObj->impl_).layout_.ROOTReadStreamer(static_cast<Leaf<i> const&>(onfileImpl).layout_);
+      static_cast<Leaf<i>&>(onfileImpl).layout_.ROOTStreamerCleaner();
+    });
+  }
+
+private:
+  std::optional<Buffer> buffer_;  //!
+  Implementation impl_;           // (serialized: this is where the layouts live)
+};
+
+// Singleton case does not need to be aliased. A special template covers it.
```

Comment:
Would it, in principle, be feasible to implement the "singleton case" with `PortableHostMultiCollection`?

Reply:
I understood this was tried and didn't work.